### PR TITLE
feat: [EXT-2887] add AppSignedRequest & AppSigningSecret

### PR DESCRIPTION
**Pull request by Jwhiles**
## Summary
This PR adds the signed request create endpoint to the API

I also added signing secret create to allow me to write an integration test for the request signing 🆒 

## Description
I tried to match the existing style

## Motivation and Context
Increase the amount of app framework features in the SDK 

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [x] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

**Comment by Jwhiles** [comment]
not sure why there are so many differences in the package lock....

**Comment by andipaetzold** [comment]
> not sure why there are so many differences in the package lock....

You upgraded the `package-lock.json` to `lockfileVersion` 2 (npm 7)

File: test/integration/app-signed-request-integration.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  createTestSpace,
+} from '../helpers'
+
+describe('AppSignedRequest api', function () {
+  let appDefinition
+  let environment
+  let space
+  let client
+
+  before(async () => {
```
Comment (andipaetzold) [inline test/integration/app-signed-request-integration.ts:17]: Please add an `after` hook to clean up everything

- Uninstall App
- Remove Environment
- Delete Space
- Delete AppDefinition

File: test/integration/app-signing-secret-integration.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import { expect } from 'chai'
+import { before, describe, test } from 'mocha'
+import { initPlainClient, getTestOrganization } from '../helpers'
+
+describe('AppSigningSecret api', function () {
+  let appDefinition
+  let client
+  let organization
+
+  before(async () => {
```
Comment (andipaetzold) [inline test/integration/app-signing-secret-integration.js:10]: please clean up

File: test/unit/entities/app-signed-request.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { cloneMock } from '../mocks/entities'
+import setupHttpMock from '../mocks/http'
+import { wrapAppSignedRequest } from '../../../lib/entities/app-signed-request'
+import { entityWrappedTest } from '../test-creators/instance-entity-methods'
+import { describe, test } from 'mocha'
+
+function setup(promise) {
+  return {
+    httpMock: setupHttpMock(promise),
+    entityMock: cloneMock('assetKey'),
```
Comment (andipaetzold) [inline test/unit/entities/app-signed-request.ts:10]: You are using the asset mock instead of a mocked signed request

**Review by andipaetzold — CHANGES_REQUESTED** [review]

File: lib/adapters/REST/endpoints/app-signing-secret.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  http: AxiosInstance,
+  params: GetAppDefinitionParams
+) => {
+  return raw.get<AppSigningSecretProps>(
+    http,
+    `/organizations/${params.organizationId}/app_definitions/${params.appDefinitionId}/signing_secret`
+  )
+}
+
+export const create: RestEndpoint<'AppSigningSecret', 'create'> = (
```
Comment (andipaetzold) [inline lib/adapters/REST/endpoints/app-signing-secret.ts:20]: Why having two identical endpoints? (`create` and `update`)? There is no difference, right?
Should they maybe be combined? Something similar to `upsert` maybe.

Comment (Jwhiles) [reply lib/adapters/REST/endpoints/app-signing-secret.ts:20]: yeah, true. I didn't notice that we have this 'upsert' pattern - I'll change to that.

**Comment by andipaetzold** [comment]
@Jwhiles Can you adjust the title to something like `feat: add AppSignedRequest & AppSigningSecret`.

`Add create` doesn't really say what this change does.

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**